spec/helpers/test-decoration-layer.coffee

```diff
@@ -19,7 +19,7 @@ class TestDecorationLayer
       @markerIndex.insert(markerId, Point.fromObject(rangeStart), Point.fromObject(rangeEnd))
       @tagsByMarkerId[markerId] = tag
 
-    @buffer?.preemptDidChange(@bufferDidChange.bind(this))
+    @buffer?.registerTextDecorationLayer(this)
```

Comment:
Alternatively we might register just the handler function instead of the whole text decoration layer object. Still, it feels somewhat convenient to have access to the entire object and reads better than `registerTextDecorationLayerChangeHandler` or `addTextDecorationLayerChangeHandler`.